citesphere/src/test/java/edu/asu/diging/citesphere/core/service/impl/AuthorityServiceTest.java

```diff
@@ -196,4 +198,37 @@ public void test_findByUri_noResults() {
         List<IAuthorityEntry> actual = managerToTest.findByUri(user, uri);
         Assert.assertTrue(actual.isEmpty());
     }
+    
+    @Test
+    public void test_create() {
+        String username = "user";
+        OffsetDateTime createdOn = OffsetDateTime.now();
+        AuthorityEntry entry = new AuthorityEntry();
+        IUser user = new User();
+        user.setUsername(username);
+        Mockito.when(entryRepository.save(entry)).thenReturn(entry);
+        try {
+            Thread.sleep(100);
+        } catch (InterruptedException e) {}
+        IAuthorityEntry actualEntry = managerToTest.create(entry, user);
+        Assert.assertEquals(username, actualEntry.getUsername());       
+        Assert.assertTrue(createdOn.isBefore(actualEntry.getCreatedOn()));
+        Mockito.verify(entryRepository).save(entry);
```

Comment:
since the saved object gets an ID when being saved I believe, you should make sure it has an ID after being saved but not before.

Reply:
It just edits values of an authority entry. So it will search using the id which is already there.

Reply:
The method is called 'test_create' which implies that it creates something. Since your resting the authorityService, I assume it creates authorities? how can that search for id?

Reply:
It just attaches the user and time to an entry and saves in the database.